awxkit/test/api/pages/test_base.py

```diff
@@ -0,0 +1,59 @@
+from unittest.mock import Mock, PropertyMock
+
+from http.client import NOT_FOUND
+import pytest
+
+from awxkit.api.pages import Base
+from awxkit.config import config
+
+
+@pytest.fixture(autouse=True)
+def setup_config():
+    config.credentials = {"default": {"username": "foo", "password": "bar"}}
+    config.base_url = ""
```

Comment:
Make sure that the patching is undone once testing this module is complete. We should be careful not to create side effects for any other tests:
```suggestion
def setup_config(monkeypatch: pytest.MonkeyPatch) -> None:
    monkeypatch.setattr(
        "credentials",
        {"default": {"username": "foo", "password": "bar"}},
    )
    monkeypatch.setattr("base_url", "")
```